### PR TITLE
Add FileConsumer superclass with Bueno and CSV

### PR DESCRIPTION
**Pull request by DanielRJohnson**
Creates a FileConsumer superclass that tracks file absolute path and file hash. Also adds a basic CSV Plugin and moves the Bueno Plugin to be under FileConsumer.

Still some things to think about:
 - Can we use hash to disallow adding duplicates
 - Reading "multi-row" data like CSVs in leads to `add_row` adding multiple rows, maybe a rename is needed

### CODE REVIEW

**Comment by qwofford** [comment]
resolves #59

File: dsi/plugins/file_consumer.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    def __init__(self, filename):
+        super().__init__()
+        self.filename = filename
+        self.file_info = OrderedDict()
+        self.file_info['abspath'] = abspath(filename)
+        sha = sha1(open(filename, 'rb').read())
+        self.file_info['sha1'] = sha.hexdigest()
+
+
+class CSV(FileConsumer):
```
Comment (qwofford) [inline dsi/plugins/file_consumer.py:24]: Nitpick...but should this be `Csv` rather than `CSV`?

**Comment by qwofford** [comment]
What do we have to do to get this PR in main still?

**Comment by DanielRJohnson** [comment]
I believe the CSV plugin adding more rows than other plugins leaves Terminal.active_metadata malformed still, needing padding of None's or forward-fill. Will look into that now

**Comment by DanielRJohnson** [comment]
Added None packing in Terminal.transload, other strategies can be added later if helpful. See changes in transload and new test in test_file_consumer.

**Review by qwofford — APPROVED** [review]
Passing pytests, docs up to date.

**Pull request merged**